sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/fragment/molecules/MultiColumnRowViewFragment.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.sample_compose_fragments.presentation.fragment.molecules
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.ui.platform.ComposeView
+import androidx.navigation.fragment.navArgs
+import dagger.hilt.android.AndroidEntryPoint
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+import uk.gov.hmrc.components.compose.ui.theme.rememberWindowSizeClass
+import uk.gov.hmrc.sample_compose_fragments.presentation.screens.molecules.MultiColumnRowViewScreen
+import uk.gov.hmrc.sample_compose_fragments.presentation.screens.sampletemplate.HmrcSurface
+
+@AndroidEntryPoint
+class MultiColumnRowViewFragment : Fragment() {
+
+    private val arg : MultiColumnRowViewFragmentArgs by navArgs()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        return ComposeView(requireContext()).apply {
```

Comment:
Although this is correct, the main goal for our "over-engineered sample" app here is to try and provide a good example of how we might integrate compose with our existing fragment/xml codebase to ease migration. So for the other components we've used a very simple xml file (`R.layout.fragment_compose_example`) which just contains a compose view in a linear layout to use as an example. We can access it with the binding like we currently do in our main app. I think it'd be nice to keep that consistent for now (although presumably when we do get to the point to start adding brand new screens in the main app with compose we may be able to do it this way, so there may be some value in having both quite soon.. for now I think consistency aids learning). Happy to hear thoughts if you disagree! :)

Reply:
Yeah, I got your point.